presto-main/src/test/java/com/facebook/presto/block/TestLazySliceArrayBlock.java

```diff
@@ -81,6 +86,29 @@ public void testDictionaryBlock()
         assertEquals(block.getSizeInBytes(), expectedSizeInBytes);
     }
 
+    @Test
+    public void testDictionaryBlockGetRegion()
+            throws Exception
+    {
+        Slice[] expectedValues = createExpectedValues(3);
+        int[] ids = new int[] { 0, 2, 1, 0, 0, 0, 1, 1, 1, 0, 1, 2 };
+        boolean[] isNull = new boolean[ids.length];
+        isNull[2] = true;
+
+        TestDictionaryLazySliceArrayBlockLoader loader = new TestDictionaryLazySliceArrayBlockLoader(expectedValues, ids, isNull);
+        LazySliceArrayBlock block = new LazySliceArrayBlock(ids.length, loader);
+
+        Block region = block.getRegion(0, 3);
+        assertFalse(region.isNull(0));
+        assertFalse(region.isNull(1));
+        assertTrue(region.isNull(2));
+
+        assertTrue(region instanceof DictionaryBlock);
```

Comment:
Add a comment explaining what this is testing. It seems to check that the numbers are re-assigned so we go from `0 2 1` to `0 1 2` where the new `2` is null in the dictionary because it never has a value in the region.